src/App.jsx

```diff
@@ -14,33 +16,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const SORT_ALPHABETICALLY = 'Sort alphabetically';
+  const SORT_BY_LENGTH = 'Sort by length';
+  const REVERSE = 'Reverse';
+  const RESET = 'Reset';
+  const [sortField, setSortField] = useState(null);
+  const [reverse, setReverse] = useState(false);
+  let visibleGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  switch (sortField) {
+    case SORT_ALPHABETICALLY:
+      visibleGoods = [...goodsFromServer].sort((a, b) => (a > b ? 1 : -1));
+      break;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    case SORT_BY_LENGTH:
+      visibleGoods = [...goodsFromServer].sort(
+        (good1, good2) => good1.length - good2.length,
+      );
+      break;
+    default:
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reverse) {
+    visibleGoods = [...visibleGoods].reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_ALPHABETICALLY)}
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== SORT_ALPHABETICALLY,
+          })}
+        >
+          {SORT_ALPHABETICALLY}
+        </button>
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_BY_LENGTH)}
+          className={classNames('button', 'is-success', {
+            'is-light': sortField !== SORT_BY_LENGTH,
+          })}
+        >
+          {SORT_BY_LENGTH}
+        </button>
+        <button
+          type="button"
+          onClick={() => {
+            setReverse(!reverse);
+            if (sortField === REVERSE) {
+              setSortField(null);
+            } else if (sortField === null) {
+              setSortField(REVERSE);
+            }
```

Comment:
The handling of the `REVERSE` sort field is unnecessary. Since the `reverse` state already manages the list reversal, you can simplify your code by removing this logic. This will prevent confusion and make the code cleaner.